exonum-java-binding/core/rust/src/utils/resource_manager/imp.rs

```diff
@@ -74,10 +72,7 @@ fn add_handle_impl<T: 'static>(handle: Handle, ownership: HandleOwnershipType) {
 fn remove_handle_impl<T: 'static>(handle: Handle, ownership: HandleOwnershipType) {
```

Comment:
This method is not correctly synchronized, as it won't detect concurrent double-free:
```
  T1       T2
  c->OK
           c->OK
  r->removed
           r->nothing to remove (OK)
```